### PR TITLE
[1068] Showing latest list of applications

### LINKED ISSUE #1068: Export flows - Cannot find the recently created flows under the list
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**
When I create a new flow in the application details page, I cannot find it in the flow list while exporting the flows.

image

In the above illustration, the flows are present but the popover shows no flows to select.

**Minimal reproduction of the problem with instructions**
1. In application details page, create a new flow
2. Click export flows
3. You should notice that the list does not show the newly created flow

* **Flogo version:** 0.9.0

### PR DESCRIPTION
**Pull request by apandurangi**
fixes #1068

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: TBD
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
When I create a flow or delete a flow in the application details page and try to export the application I do not see the latest set of resources

**What is the new behavior?**
The flows list for the Export flow is loaded once when we launch application details page and never updated. The list is now taking the recent set of resources.

### CODE REVIEW

File: apps/client/src/flogo/app/core/apps.service.ts [hunk 1]
```diff
@@ -139,6 +183,49 @@ export class AppDetailService {
...
+    return this.triggersState$.pipe(
+      take(1),
+      map(triggers => {
+        const appHasTriggers = triggers && triggers.length > 0;
+        return !!appHasTriggers;
+      })
+    );
+  }
+
+  public getAvailableShimBuildOptions$(): Observable<ShimBuildOptions[]> {
```
Comment (fcastill) [inline apps/client/src/flogo/app/core/apps.service.ts]: we only use suffix `$` to indicate variables that are streams, it's not necessary for functions

File: apps/client/src/flogo/app/core/apps.service.ts [hunk 2]
```diff
@@ -139,6 +183,49 @@ export class AppDetailService {
...
+    const triggersUsed$: Observable<Set<string>> = this.groupsByTrigger$.pipe(
+      map(flowGroups => {
+        return new Set(
```
Comment (fcastill) [inline apps/client/src/flogo/app/core/apps.service.ts]: can also use Set's chainable interface directly and avoid the creation of the array:

```ts
 return flowGroups.reduce(
    (refsSet, group) =>
      group.trigger ? refsSet.add(group.trigger.ref) : refsSet,
    new Set()
  );
```

File: apps/client/src/flogo/app/core/apps.service.ts [hunk 3]
```diff
@@ -35,8 +52,35 @@ export class AppDetailService {
...
+      map(([triggers, resources]) => groupByResource(triggers, resources))
+    );
+
+    this.isEmpty$ = this.resourcesState$.pipe(
+      map(resources => !resources || resources.length <= 0),
+      shareReplay(1)
+    );
+  }
+
+  private get triggersState$() {
```
[the remark this answers is not in the document]
Comment (apandurangi) [reply apps/client/src/flogo/app/core/apps.service.ts]: triggers$ is already defined in `app-resources-state.service.ts` and we are returning the same in `apps.service.ts`

I used the name to specify that the this is from the state maintenance service.

I will implement the review comment.

**Comment by apandurangi** [comment]
@fcastill Implemented the review comments.

**Pull request merged**